### PR TITLE
Fix py-notebook for Python2

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-notebook/package.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ class PyNotebook(PythonPackage):
     depends_on('py-nbformat', type=('build', 'run'))
     depends_on('py-nbconvert', type=('build', 'run'))
     depends_on('py-ipykernel', type=('build', 'run'))
-    depends_on('py-ipykernel@5.1.0:', when='@4.2.0:', type=('build', 'run'))
+    depends_on('py-ipykernel@5.1.0:', when='@4.2.0: ^python@3:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-notebook/package.py]: This is required for both Python 2 and 3

File: var/spack/repos/builtin/packages/py-notebook/package.py [hunk 2]
```diff
@@ -47,3 +47,8 @@ class PyNotebook(PythonPackage):
     depends_on('py-jupyter-client@5.3.1:', type=('build', 'run'), when='@6:')
     depends_on('py-terminado@0.8.1:', type=('build', 'run'), when='@6:')
     depends_on('py-ipaddress', type=('build', 'run'), when='@6: ^python@:2.8')
+
+    @when('@5.7.8')
+    def install_args(self, spec, prefix):
+        args = ['--prefix={0}'.format(prefix), '--root=/']
+        return args
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-notebook/package.py]: This version is not in Spack, and these args are already added automatically

File: var/spack/repos/builtin/packages/py-notebook/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-notebook/package.py]: As far as I can tell, no. E.g., for `notebook==5.7.8` pip installs `ipykernel-4.10.1`:
```console
$ mktmpenv -p python2
(...)
Collecting ipykernel
  Downloading https://files.pythonhosted.org/packages/59/9e/e16335ee2d645ee48f082e4207d4fad9bdce09cc1537e76320c341d8d75c/ipykernel-4.10.1-py2-none-any.whl (109kB)
(...)
```

Comment (iarspider) [reply var/spack/repos/builtin/packages/py-notebook/package.py]: Actual requirements for `notebook==5.7.8`:
```
Requires-Dist: jinja2
Requires-Dist: tornado (<7,>=4.1)
Requires-Dist: pyzmq (>=17)
Requires-Dist: ipython-genutils
Requires-Dist: traitlets (>=4.2.1)
Requires-Dist: jupyter-core (>=4.4.0)
Requires-Dist: jupyter-client (>=5.2.0)
Requires-Dist: nbformat
Requires-Dist: nbconvert
Requires-Dist: ipykernel
Requires-Dist: Send2Trash
Requires-Dist: terminado (>=0.8.1)
Requires-Dist: prometheus-client
Requires-Dist: ipaddress ; python_version == "2.7"
```

File: var/spack/repos/builtin/packages/py-notebook/package.py (hunk 2 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-notebook/package.py]: With the default arguments, the install fails:
```
==> Executing phase: 'install'
==> [2019-12-11-11:49:11.251112] '/mnt/build/jenkins/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/python-2.7.16-thcifv7hiw2y5pgh5isqo4oodmunmvak/bin/python2.7' '-s' 'setup.py' '--no-user-cfg' 'install' '--prefix=/mnt/build/jenkins/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/py-notebook-5.7.8-l4usuot3ns4f2mwul47zsli35aku2d6a'
running install
running bdist_egg
Aborting implicit building of eggs. Use `pip install .` to install from source.
```
Alternatively, adding a dependency on `setuptools` fixes the error.

**Comment by iarspider** [comment]
Updated version ranges of all dependencies to what is found inside `setup.py` files.

**Pull request merged**